Gems/Multiplayer/Code/Include/Multiplayer/IMultiplayerDebug.h

```diff
@@ -7,9 +7,70 @@
  */
 
 #pragma once
+#include <AzCore/Math/MathStringConversions.h>
 
 namespace Multiplayer
 {
+    //! Categories for Auditing logs used in MultiplayerDebugAuditTrail
+    enum class MultiplayerAuditCategory
+    {
+        Desync,
+        Input,
+        Event
+    };
+
+    //! @class IMultiplayerAuditingDatum
+    //! @brief IMultiplayerAuditingDatum provides an interface for auditing datums of a given auditing event
+    class IMultiplayerAuditingDatum
+    {
+    public:
+        virtual ~IMultiplayerAuditingDatum() = default;
+        virtual IMultiplayerAuditingDatum& operator= (const IMultiplayerAuditingDatum&) { return *this; }
+
+        //! Retrieves the name of the auditing datum
+        virtual const AZStd::string& GetName() const = 0;
+
+        //! Retrieves the Client and Server values of the datum as strings
+        virtual AZStd::pair<AZStd::string, AZStd::string> GetClientServerValues() const = 0;
+
+        //! Clones the Datum to a new unique_ptr
+        virtual AZStd::unique_ptr<IMultiplayerAuditingDatum> Clone() = 0;
+    };
+
+    //! @struct MultiplayerAuditingElement
+    //! @brief MultiplayerAuditingElement contains a list of datums for a given auditing event
+    struct MultiplayerAuditingElement
+    {
+    public:
+        MultiplayerAuditingElement() = default;
+        MultiplayerAuditingElement(const MultiplayerAuditingElement& rhs)
```

Comment:
Did we consider move assignment operator for struct to avoid deep clones? or is there always a reason for deep clones?

Reply:
The usage of unique_ptr in the elements vector (which allows it to function as an abstract value vector) forces it.